test/fail_compilation/retscope2.d

```diff
@@ -291,5 +291,28 @@ struct T17388
     return t.foo();
 }
 
+/****************************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope2.d(1306): Error: returning `& i` escapes a reference to local variable `i`
```

Comment:
I would strongly prefer if the error message didn't have `return` in it when there are no `return` statements to be found.
For reference, if you uncomment the constructor, it looks like this:
```
Error: reference to local variable `i` assigned to non-scope parameter `p` calling `scope_bug17370.Struct.this`
```

Ideally the error message should look like this:
```
Error: reference to local variable `i` assigned to heap-allocated `S1300.oops` with possibly longer lifetime
```

Reply:
While I agree with you, improving the message is a minor issue relative to fixing this bug. It requires some refactoring of the code to make it work. I suggest adding it as an issue on bugzilla so it can be addressed on its own.

Reply:
Ok, works for me: https://issues.dlang.org/show_bug.cgi?id=17784.